File: 10dayOfStatisc/day0.1.js
function getMedian(input) {
    let median = 0
        // get the median
        // [x1, x2, x3, x4, x5, xn]
    const pos = Math.round(input.length / 2) // media value in the array
    if (input.length % 2 != 0) {
        // for the odd length
        // median = x3
        median = input[pos - 1]
    } else { // for the par one
        // [x1, x2, x3, x4, x5, x6]
        // mediam = (x3+x4) / 2
        const pos_2 = pos - 1 // the first median value. pos is th second media value
        const x3 = input[pos]
        const x4 = input[pos_2]
        median = (x3 + x4) / 2
    }
    return median.toFixed(1)
}

function processData(input) {
    //Enter your code here
    var mean = 0;
    var mode = -1;
    var mode_dict = {}

    // ordenar o array
    for (let i = 0; i < input.length; i++) {
        let pos = i
        let menor = input[i]
        for (let j = i; j < input.length; j++) {
            if (input[j] < menor) {
                menor = input[j]
                pos = j
            }
        }
        let aux = input[i]
        input[i] = menor
        input[pos] = aux
            // sum the N
        mean += input[i]

        // get the most popular one
        const key = `${input[i]}`
        if (mode_dict[key] != undefined) {
            mode_dict[key] += mode_dict[key]
        } else {
            mode_dict[key] = 1
        }
    }


    //get the mean
    // [x1, x2, x3, x4, x5, xn]
    // mean = (x1+x2 .. +xn)/ n
    const n = input.length
    mean = (mean / n).toFixed(1)

    // get the media
    const median = getMedian(input)

    // the most popular one
    const most_popular_one = getMode(input, mode_dict)
    console.log(mean)
    console.log(median)
    console.log(most_popular_one)
}

function getMode(input, mode_dict) {
    var most_popular_one = input[0] // popular incializations
    let largest = -1 // largest inicializations
    for (let i = 0; i < input.length; i++) {
        const item_key = input[i] // the one who must be popular
        const apears = mode_dict[item_key] // time appears
        if (apears > largest) {
            largest = apears
            most_popular_one = item_key
        }
    }
    return most_popular_one
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function(input) {
    _input += input;
});

process.stdin.on("end", function() {
    processData(_input);
});

processData([64630, 11735, 14216, 99233, 14470, 4978, 73429, 38120, 51135, 67060])

File: 10dayOfStatisc/day0.1.py
# Enter your code here. Read input from STDIN. Print output to STDOUT


def get_the_mean(array: list):
    sum = 0
    for element in array:
        sum += element
    return float(sum/len(array))


#[1, 2, 3, 4, 5]

def get_the_median(array: list):
    median_pos = int(len(array)/2)
    if not (len(array) % 2 == 0):
        return array[median_pos]

    median_pos_firt = median_pos-1
    median_pos_sec = median_pos

    median = (array[median_pos_firt] + array[median_pos_sec]) / 2
    return median


def get_the_mode(array: list):
    verified_numers = []
    maior = -1
    popular_one = -1
    for element in array:
        if element not in verified_numers:
            count = array.count(element)
            verified_numers.append(element)

            if maior < count:
                maior = count
                popular_one = element

    return popular_one



if __name__ == '__main__':
    n = int(input())
    lista= list(map(int, input().rstrip().split()))
    
    
    lista = sorted(lista)
    mean = get_the_mean(lista)
    median = get_the_median(lista)
    mode = get_the_mode(lista)
    print('%.1f' % round(mean, 1))
    print('%.1f' % round(median, 1))
    print(mode)
